Release v6.31

MichaIng opened the pull request:
https://github.com/MichaIng/DietPi/blob/dev/CHANGELOG.txt

MichaIng committed:
META | DietPi-Survey_report: Add v6.30 and v6.31 support

MichaIng committed:
v6.31

+ DietPi-Software | Code fixes and enhancements, based on CodeFactor scan

MichaIng committed:
v6.31

+ DietPi-Patch | Minor code enhancements based on CodeFactor scan

MichaIng committed:
v6.31

+ DietPi-Boot | Run network time sync before estimating network details. Waiting for network only checks for a default route, but especially in case of DHCP this can be present before a valid IP has been assigned. Network time sync is done anyway and, if not a local NTP server or custom method has been chosen, serves as an additional prove for internet connectivity. The non-interactive DietPi-Run_NTPD call does not require or use /run/dietpi/.network at all, hence no regression is implied. All services which require internet connectivity, especially those which read /run/dietpi/.network, should start "After=dietpi-boot.service", those which read /run/dietpi/.network additionally should call "ExecStartPre=/boot/dietpi/func/obtain_network_details" to have updated network info, e.g. on service restarts (ToDo).

MichaIng committed:
v6.31

+ DietPi-Software | GMediaRender/WireGuard: Update network info before reading info file on service start. This prevents invalid IP info if DHCP lease on boot finishes before obtain_network_details is called or if during session the device is connected to a different network or DHCP applies a new IP and services are (re)started afterwards. Planned long-term solution is a set of functions to read network info directly from system instead of relying on correct /run/dietpi/.network content (ToDo).

MichaIng committed:
v6.31

+ CHANGELOG | Add v6.31 changelog
+ CHANGELOG | GMediaRender+WireGuard: Resolved an issue where service start could have failed due to invalid network information. Many thanks to @fnsnyc for reporting this issue: https://github.com/MichaIng/DietPi/issues/3519
+ CHANGELOG | After travelling back to the future 2030, we were able to apply bug fixes with alien technology to the v6.30 release back in 2020.

MichaIng committed:
v6.31

+ DietPi-Fan_control | Minor coding enhancements

MichaIng committed:
v6.31

+ DietPi-Backup | Minor coding enhancements

MichaIng committed:
v6.31

+ DietPi-Cleaner | Minor coding enhancements

MichaIng committed:
v6.31

+ DietPi-Cloudshell | Minor coding enhancements

MichaIng committed:
v6.31

+ DietPi-Config | Minor coding enhancements

MichaIng committed:
v6.31

+ DietPi-Drive_Manager | Tiny coding

MichaIng committed:
v6.31

+ DietPi-LED_control | Minor code enhancements

MichaIng committed:
v6.31

+ DietPi-Services | Minor coding enhancements

MichaIng committed:
v6.31

+ DietPi-Survey | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Change_hostname | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Benchmark | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Globals | Minor code enhancements

MichaIng committed:
v6.31

+ DietPi-Set_software | Minor code enhancements

MichaIng committed:
v6.31

+ DietPi-WiFiDB | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Arr_to_RAM | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Pre-patches | Satisfy CodeFactor

MichaIng committed:
v6.31

+ DietPi-Drive_Manager | Minor code enhancement

MichaIng committed:
v6.31

+ DietPi-Drive_Manager | Tiny code enhancement

MichaIng committed:
META | Add CodeFactor to DietPi source section and bash to 3rd party list

MichaIng committed:
v6.31

+ DietPi-PREP | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Software | Tiny code enhancement

MichaIng committed:
v6.31

+ DietPi-Software | Always call curl with "-f" option to have it exiting with error code and meaningful short error message in case of webserver error responses, instead of printing the HTML error page with exit code zero.

CedArctic committed:
v6.31 (#3471)

+ DietPi-Software | Add Firefox Sync Server

MichaIng committed:
v6.31 (#3531)

+ DietPi-Software | rTorrent: Set default pieces.memory.max.set to total physical memory size in MiB as intended
+ DietPi-Software | rTorrent: Fix invalid dht.mode.set value, which must be "on" instead of "enable"

MichaIng committed:
v6.31

+ DietPi-Patch | Fix variable call

MichaIng committed:
v6.31

+ DietPi-Imager | Store original directory path, before navigating into temporary working dir, so dietpi-explorer can start at intended location

MichaIng committed:
v6.31

+ DietPi-Imager | We must navigate to origin dir, otherwise the resulting image is removed with /tmp/$G_PROGRAM_NAME on exit

MichaIng committed:
v6.31

+ DietPi-PREP | Install rng-tools5 as modern entropy daemon wherever it has been proven to work. Separate entropy daemon and initramfs package estimation for allow more flexible adjustments based on further tests.

MichaIng committed:
v6.31

+ DietPi-PREP | Install rng-tools5 on all RK3399 SBCs

MichaIng committed:
v6.31 (#3536)

+ DietPi-Services | Add support for new Firefox Sync Server
+ DietPi-Software | Firefox Sync Server: Add online docs URL: https://dietpi.com/phpbb/viewtopic.php?p=24713#p24713
+ DietPi-Software | Firefox Sync Server: Do not edit Makefile to use our config, instead call gunicorn with correct option directly
+ DietPi-Software | Firefox Sync Server: Add strict service hardening to allow R/W only to our config+database dir, the ffsync users home
+ DietPi-Software | Firefox Sync Server: Do not enable service on install, since it is now controlled by DietPi-Services
+ DietPi-Patch | Make userdata dir world-executable so service users don't need to be in dietpi group to access their data dir: https://github.com/MichaIng/DietPi/pull/3536#issuecomment-628515444

MichaIng committed:
v6.31

+ DietPi-Software | Deluge: Enhance uninstall to assure all service and user traces are removed

MichaIng committed:
v6.31

+ README | Add Syncthing and Firefox Sync Server to 3rd party sources list

MichaIng committed:
v6.31

+ DietPi-Set_hardware | Tiny

MichaIng committed:
v6.31

+ DietPi-Config | Add Ethernet link speed selection to Network>Ethernet menu. The dietpi-set_hardware option and dietpi.txt entry exists for a long time, but it was only exposed as first run setup option.

MichaIng committed:
v6.31

+ CHANGELOG | Added Ethernet link speed selection to Network>Ethernet menu

MichaIng committed:
v6.31

+ DietPi-Config | Fix Ethernet link speed menu if no link speed force option is supported by the network adapter

MichaIng committed:
v6.31

+ Odroid XU4 | Update boot.ini: Compact HDMI-PHY control settings and expose "new" (2017) DDR frequency as an optional setting: https://github.com/hardkernel/u-boot/commit/edb23d4ba7c402485929aa2777aeea67007b23b5

MichaIng committed:
v6.31

+ DietPi-Imager | Error handle some more commands which allows to retry and adjust them in case of error, before exiting

MichaIng committed:
v6.31

+ DietPi-PREP | Install fdisk on Buster+ systems, where it has become an own package. On Bullseye it has been degraded from "required" to "important", which means that it would be autoremoved otherwise. This is used by DietPi-FS_partition_resize, parted cannot be used since it does not allow to resize a partition non-interactively, currently, and does not support max resizing but only to a defined end, which must be estimated first.
+ DietPi-PREP | Install gdisk on systems where the root file systems is on a GPT partition, used by DietPi-FS_partition_resize
+ DietPi-PREP | Fix "fat" file system detection it the it is the first file system on the list. This is a failsafe step only, since those are reported as "vfat" in all known cases.

MichaIng committed:
v6.31

+ DietPi-Imager | Remove obsolete exit calls, replace Exit_On_Fail with direct exit call and remove loopback device via G_EXIT_CUSTOM exit trap.

MichaIng committed:
v6.31

+ DietPi-Imager | Fix example about how to limit 7zr threads

MichaIng committed:
v6.31 (#3545)

+ Odroid C4 | Add initial support

MichaIng committed:
v6.31

+ DietPi-FirstBoot | Apply static DNS nameserver via /etc/resolv.conf directly, since resolvconf is not preinstalled anymore: https://github.com/MichaIng/DietPi/commit/22131fe4b63124bf9e93a229f45fc49380dd09b4#diff-0e127c4cad3f8401a925d13d05b5ae87

MichaIng committed:
v6.31

+ DietPi-Imager | Remove the loopback device only if a file is attached, since the device file exists always

MichaIng committed:
v6.31

+ DietPi-Explorer | Print some info when processing the content of the current dir, since this ("wc -c" on all files) can take some seconds on slower drives and/or when large files are contained. ToDo: Think about scraping "ls -Al" output instead!

MichaIng committed:
v6.31

+ DietPi-Software | Use new method clear terminal by moving existing content into history, thus preserving it

MichaIng committed:
v6.31

+ DietPi-PREP | Revert /tmp tmpfs size to default after it has been set by DietPi-Set_swapfile

MichaIng committed:
v6.31

+ DietPi-PREP | Fix check for GPT partition table

MichaIng committed:
v6.31

+ DietPi-Globals | G_OBTAIN_CPU_TEMP: Add support for new Odroid C1 Armbian legacy Linux 5.4.40 image: https://dietpi.com/phpbb/viewtopic.php?p=24860#p24860
+ DietPi-Globals | Minor coding enhancements which i.a. removes the need to pre-create G_PROMPT_BACKUP_DISABLED 
 and G_THREADING_ENABLED variables by directly checking against the default value

MichaIng committed:
v6.31 (#3546)

+ DietPi-Software | Folding@Home: Update to new v7.6.X
+ DietPi-Software | Folding@Home: Create default config file, prevent config backup on each change
+ DietPi-Software | Folding@Home: Disable file logging by default, "journalctl -u fahclient" can be used instead, reduce logging level to minimum
+ DietPi-Software | Folding@Home: Pause on initial service start. When folding is manually started, this is overridden per-slot for next service start but prevents us from pulling a work unit even on test installs.
+ DietPi-Software | Folding@Home: The previous package does not need to be purged anymore before the new one can be installed
+ DietPi-Software | Folding@Home: It is now possible to prevent the service start on package install
+ DietPi-Software | Folding@Home: FAHClient will not store given command line options to a new config file anymore, hence we need to create our own default config but can skip all command line options within the systemd unit
+ DietPi-Software | Folding@Home: Fix signing out from all work units on uninstall
+ DietPi-Patch| Folding@Home: Reinstall to apply v7.6.X update, remove obsolete files and purge installed package, since the old one caused issues on reinstall

MichaIng committed:
v6.31

+ DietPi-Patch | Assure that correct permissions for /run/dietpi and /run/dietpi/.network are applied when creating them (pre-v6.29)

MichaIng committed:
v6.31

+ DietPi-Software | Plex Media Server: On uninstall, remove systemd unit which currently survives the package purging: https://github.com/MichaIng/DietPi/issues/3551

MichaIng committed:
v6.31

+ DietPi-PREP | Install modern rng-tools5 on Odroid XU4 as well: https://github.com/MichaIng/DietPi/issues/3544#issuecomment-632086158

MichaIng committed:
v6.31

+ DietPi-Cloudshell | Calling the external "clear" binary is quite slow. It is possible to redirect its output to use it later with only shell-internal commands, which is orders of magnitudes faster. We could use known escape sequences here but using the sequence the "clear" estimates to be the best for the current terminal type makes it most compatible and does not change the current behaviour in any way, besides performance ;).
+ DietPi-Cloudshell | Minor coding enhancement

MichaIng committed:
v6.31

+ DietPi-PREP | ROCK Pi S: rockpis-dtbo is not required as it doubles the overlays that are already provided (among others) with the kernel package

MichaIng committed:
v6.31

+ DietPi-PREP | Do not remove /home (recreated anyway) and /media but their content only, since their existence is required to fulfil FHS and their absence can hence lead to issues: https://github.com/MichaIng/DietPi/issues/3558, https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=956495#15

MichaIng committed:
v6.31

+ DietPi-Software | Sonarr: Install from stable/master branch instead of develop
+ DietPi-Software | Sonarr/Radarr/Lidarr: Add /media to ReadWritePaths to not break instances where users use /media for their mounts: https://github.com/MichaIng/DietPi/issues/3561
+ DietPi-Software | Sonarr/Radarr/Lidarr/Jackett: When enabling services for systemd-controlled startup at boot, wait for dietpi-boot.service which is a better measure to ensure network is up and time sync as well, required for any encrypted network access where certificates timestamps must match.

MichaIng committed:
v6.31 (#3565)

+ DietPi-Logclear | Resolve an issue where .db-shm and .db-wal files are were not excluded from processing, bug introduced with v6.29

MichaIng committed:
v6.31

+ DietPi-Patch | Assure /media exists to fulfil FHS, fix htpdate service startup and probably other issues: https://github.com/MichaIng/DietPi/issues/3558

MichaIng committed:
v6.31

+ CHANGELOG | Resolved potential issues due to missing /media directory on DietPi images

MichaIng committed:
v6.31

+ DietPi-Globals | Some ordering, wording and explanation enhancements
+ DietPi-Globals | Set and export $LANG as well even that its effect is fully overwritten by $LC_ALL, since some 3rd party scripts/programs might check it to derive the used/default locale

MichaIng committed:
v6.31

+ DietPi-Set_software | locale: Minor wording enhancements
+ DietPi-Set_software | locale: Show output of dpkg-reconfigure call since it might be interesting which locales are actually generated and if any warnings appear
+ DietPi-Set_software | locale: Set and export $LANG as well, since errors can appear as well if this has an invalid value, even that its effect is fully overridden by $LC_ALL

MichaIng committed:
v6.31

+ DietPi-Software | Pi-hole: Allow direct access to further local fonts types, used now by AdminLTE: https://github.com/pi-hole/pi-hole/pull/3403/files#diff-db31ccc9ffd35ede6f6bf7ba1e5c8407

MichaIng committed:
v6.31

+ DietPi-Software | Pi-hole: Minor wording in Lighttpd config

MichaIng committed:
v6.31

+ DietPi-Software | Pi-hole: Grant admin panel access to all local fonts types: https://github.com/pi-hole/pi-hole/pull/3403/files#r432926507

MichaIng committed:
v6.31

+ DietPi-Software | Pi-hole: Grant admin panel access to all local fonts types: https://github.com/pi-hole/pi-hole/pull/3403/files#r432926507

MichaIng committed:
v6.31

+ DietPi-PREP | Locales: Export $LANG as well as in case of invalid entry this causes an error message as well, even that it is effectively overridden by $LC_ALL. Export both after locale generation, before calling any other command.

MichaIng committed:
v6.31

+ DietPi-PREP | Disable dpkg state translations and skip installing recommends already on initial dependency installs

MichaIng committed:
v6.31

+ DietPi-Software | WireGuard: Enable for Odroid N2 and C4, remove obsolete Rock64 implementation, we'll add a generic Armbian solution at a later date.
+ DietPi-Software | WireGuard: The package is now available via Debian backports, hence add Bullseye repo only on RPi (there are no Raspbian backports) or Stretch.

MichaIng committed:
v6.31

+ Network | Grant all users "ping" access without the need for sudo, setuid, CAP_NET_RAW or CAP_NET_ADMIN. Further infos and discussion about this:
  - https://fedoraproject.org/wiki/Changes/EnableSysctlPingGroupRange
  - https://github.com/systemd/systemd/pull/13141
  - https://github.com/MichaIng/DietPi/issues/1012

MichaIng committed:
v6.31

+ DietPi-Patch | All users are permitted to "ping" now without any further capabilities, hence remove them for security reasons: https://github.com/MichaIng/DietPi/commit/caec92e92e136a2f731f6d4c45db8463896fc80a

MichaIng committed:
v6.31

+ CHANGELOG | "ping" can now be used by all users without any file capabilities, sudo or setuid. For this we allow all users to create native ICMP sockets which are available since Linux 3.X but disabled by default on Debian. Other distributions and systemd (upstream) have this enabled by default and for security and usability reasons we follow them: https://fedoraproject.org/wiki/Changes/EnableSysctlPingGroupRange

MichaIng committed:
v6.31

+ README.md | Load logo from GitHub resource and the new myVirtualserver logo from their website

MichaIng committed:
v6.31

+ DietPi-Software | Samba Server: Change default server-side file permissions to 0664 (no execute permissions)

MichaIng committed:
v6.31

+ CHANGELOG | Samba Server: Changed default server-side file permissions from 0775 to 0664

MichaIng committed:
v6.31

+ DietPi-Survey_report | Prevent concurrent executions

MichaIng committed:
v6.31

+ DietPi-Cron_daily | MOTD: Error out if webserver answers with 40X error instead of storing the error response in the MOTD script

MichaIng committed:
v6,31

+ DietPi-Banner | MOTD: (Re)download the MOTD not only if the file does not exist, but also if it is empty (and writable), e.g. due to a failed download before. This resolves an issue where the the first banner after a failing daily cron MOTD download did not show any MOTD. A failed download produced an empty file, hence the banner did not try a fresh download.
+ DietPi-Banner | MOTD: Initialise the $motd variable empty (local) to assure it is not e.g. exported from the parent shell

MichaIng committed:
v6.31

+ DietPi-Globals | G_CHECK_URL_ATTEMPTS: Do not allow a value of "0" which would result in a single attempt but according to dietpi.txt it is invalid and reverted to default "2".
+ DietPi-Globals | Start implementation of default gateway, network interface and IP print functions, to replace DietPi-Obtain_network_details mid-term. Since network info can change during sessions, it makes sense to always estimate the current value required with an efficient function.

MichaIng committed:
v6.31

+ DietPi-Survey_report | Fix concurrent execution detection

MichaIng committed:
v6.31

+ DietPi-Survey_report | Simplify by avoid piping and sub processes

MichaIng committed:
v6.31

+ DietPi-Survey_report | Keep track of previous output

MichaIng committed:
v6.31

+ DietPi-Globals | G_EXEC: Force usage of originating tty for STDIN, STDERR and STDOUT when spawning subshell for error investigation. This assures that bash recognises itself as interactive shell, regardless of stream redirection of originating shell/script. This could be done via "-i" as well but re-binding streams to tty also fixes some interactive programs which rely on STDOUT and STDERR being detected as tty [[ -t [12] ]], e.g. nano being unusable otherwise, when streams are redirected or piped to file or "tee". This is relevant for dietpi-update and dietpi-software on first login.

MichaIng committed:
v6.31

+ DietPi-PREP | Add u-boot to Odroid C4 requirements. N2 u-boot is currently pre-installed but not part of the C4 repo, as well it is not dependency of the C4 kernel. However lets assume the pre-installed N2 u-boot works, keep it to enable auto-upgrades once a dedicated C4 package has been added to the C4 repo.

MichaIng committed:
v6.31

+ DietPi-PREP | Force new config for vmtouch

MichaIng committed:
v6.31

+ DietPi-Imager | Automatically limit lzma2 threads on low RAM systems

MichaIng committed:
v6.31

+ DietPi-Software | On firstrun installs, flag DietPi-RAMlog for reinstall only if it is stilled marked to stay installed after all manual choices. Else this is overriding manual logging choice 0 and -3 since Apply_Logging_Choices() only touches "(( ${aSOFTWARE_INSTALL_STATE[103]} == 2 ))": https://github.com/MichaIng/DietPi/issues/3588

MichaIng committed:
v6.31

+ CHANGELOG | Logging choice: Resolved an issue where manually choosing logging options "None" and "Full" during firstrun installs did not uninstall DietPi-RAMlog as intended

MichaIng committed:
v6.31

+ DietPi-Globals | Tiny

MichaIng committed:
v6.31

+ DietPi-Drive_Manager | Move RootFS: Assure that DietPi userdata from rootfs, respectively its symlink, is copied to the target drive as well as all mountpoint directories (without content). This is achieved by excluding mounts content via tmp rootfs mountpoint and copying from there. This allows to skip excluding known tmpfs mountpoints manually and assures that source and target match 100% (aside of extended attributes and ACLs). rsync "-x" option prevents copying mounts content, but it copies permissions of mountpoint dirs according to mount options instead of those of the dir on the parent fs. This might be not wanted in cases. E.g. a shared drive or tmpfs has mounted with loose permissions to allow anyone access, but it is not wanted that everyone can access the directory below the mount point, when it is not mounted, e.g. to prevent a full rootfs and such. This might be a topic for rsync to add this behaviour by default when using "-x".

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-Drive_Manager: Resolved an issue where moving the RootFS to an external drive did not include DietPi userdata

MichaIng committed:
v6.31

+ DietPi-Login | Allow prevention of dietpi-login call by setting/exporting $G_DIETPI_LOGIN. E.g. when shell is called as subshell of G_EXEC or dietpi-login itself, we don't want autostart programs to be launched.

MichaIng committed:
v6.31

+ DietPi-Globals | G_EXEC: Prevent dietpi-login call in subshell. We do not want e.g. autostart programs to be launched, minimise overhead and the chance for nested error loops.

MichaIng committed:
v6.31

+ DietPi-Login | Prevent automated nested dietpi-login calls from subshells

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-Login: Resolved an issue where the script called itself from within the error handler subshell. Generally the error handler subshell must never call dietpi-login (and related autostart programs) and dietpi-login must never call itself from an interactive subshell. Many thanks to @nosyaardvark for reporting this issue: https://github.com/MichaIng/DietPi/issues/3583#issuecomment-643583664

MichaIng committed:
v6.31

+ DietPi-Software | WireGuard: Installing resolvconf when choosing client setup, used by wg-quick to apply DNS

MichaIng committed:
v6.31

+ CHANGELOG | Resolved an issue where WireGuard in client mode failed to start due to missing resolvconf. It is now installed together with WireGuard when choosing client setup. Many thanks to @yahoo456 for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?f=11&t=7783

MichaIng committed:
v6.31

+ DietPi-Drive_Manager | Preserve SSHFS entries in fstab

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-Drive_Manager: SSHFS entries in fstab are now detected and preserved

MichaIng committed:
v6.31

+ DietPi-Cleaner | Enhance file search performance by skip traversing /mnt if not selected for inclusion
+ DietPi-Cleaner | Minor coding and wording

MichaIng committed:
v6.31

+ DietPi-Cleaner | Add actually matched file to find_string
+ DietPi-Cleaner | Do not escape parenthesis for find command when added via variable since this is only relevant when typing it directly into the terminal. Else the backslash is interpreted literally.

MichaIng committed:
v6.31

+ DietPi-Cleaner | Reduce output length, allowing better review in case of many matches

MichaIng committed:
v6.31

+ DietPi-Globals | Tiny

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-Cleaner: Enhanced performance of the files cleaner if /mnt is skipped, especially in combination with large drives or network mounts. Many thanks to @maartenlangeveld for revealing the underlying issue: https://github.com/MichaIng/DietPi/issues/3609

MichaIng committed:
v6.31

+ DietPi-Benchmark | Minor performance enhancements

MichaIng committed:
v6.31

+ DietPi-Globals | G_OBTAIN_CPU_TEMP: Revert earlier v6.31 change: ${afp_temperature[@]} must not be quoted to allow glob expansion
+ DietPi-Globals | G_OBTAIN_CPU_TEMP: We check for G_CHECK_VALIDINT on temp before accepting it, hence float handling is not required, especially not after "(( $temp >= 200 ))" which will always be false in case of present decimal point.

MichaIng committed:
v6.31

+ DietPi-Survey_report | Enhance performance significantly by handling CPU times as integers until calculating final results in chosen scale. This allows to use native bash arithmetic instead of bc which is orders of magnitude faster. This became relevant since with Debian Buster bc execution in this environment somehow became very slow. Preserve CPU times scale and CPU temp 5 digits format check from DietPi-Benchmark respectively DietPi-Globals.

MichaIng committed:
v6.31

+ DietPi-Software | Update HAProxy to lastest stable version: https://github.com/MichaIng/DietPi/issues/3604

MichaIng committed:
v6.31 (#3605)

+ DietPi-Software | Ubooquity: Download newest version from Vae Mendis Software directly
+ DietPi-Software | Ubooquity: Make comics and ebooks dirs writable for "dietpi" group members (downloaders, file servers, ...) by default.
+ DietPi-Software | Ubooquity: Set users home dir and move logs to /var/log/ubooquity (DietPi-RAMlog). Sadly we cannot disable the log file since "journalctl -u ubooquity" doubles it.
+ DietPi-Software | Ubooquity: Use G_EXEC thoroughly to be most verbose about what we're doing and prevent unexpected situations because of failed user/dir creation and such.

MichaIng committed:
v6.31

+ DietPi-Obtain_HW_model | RPi: Always check newest unpublished commits on revision code docs and add potential compute module 4 support. Although no info about this has been published yet, the CM4 is the only model where info in upstream kernel/firmware can be found about already, so we assume it will be released next. Most importantly G_HW_MODEL=4 is detected correctly for next release, a wrong G_HW_MODEL_NAME would be visual-only.

MichaIng committed:
v6.31

+ DietPi-Software | Node.js: libatomic1 is required to start Node since last year: https://github.com/MichaIng/DietPi/issues/3614
+ DietPi-Software | Node.js: Increase install verbosity and error handling
+ DietPi-Software | Certbot: Minor wording, uninstall enhancement and use follow official notation "Certbot" with lower case "b".

MichaIng committed:
v6.31

+ CHANGELOG | Node.js: Resolved an issue where Node failed to start due to missing new libatmoic1 dependency. Many thanks to @larsno for reporting this issue: https://github.com/MichaIng/DietPi/issues/3614

MichaIng committed:
v6.31

+ DietPi-Config | Remove obsolete spaces

MichaIng committed:
v6.31

+ DietPi-Survey_report | Add percentages to counts
+ DietPi-Survey_report | Sort benchmark results by average score
+ DietPi-Survey_report | Minor coding and wording

MichaIng committed:
v6.31

+ DietPi-Survey_report | Sort CPU benchmark with shortest time (fastest CPU) on top
+ DietPi-Survey_report | Be precise on benchmark units

MichaIng committed:
v6.31 (#3617)

+ General | Add support for Raspberry Pi OS (64-bit) by separating RPi detection from Raspbian detection and adding new global variable: $G_RASPBIAN
+ DietPi-PREP | DietPi-Globals never creates .hw_model anymore, hence skip explicit script removal
+ DietPi-PREP | Since RPi can run Debian and Raspbian 32 and 64 bit images and kernel can be loaded in 64-bit mode on 32-bin image, we need to check if its a Debian or Raspbian image and if the primary deb package architecture is arm64 or not (armhf) to be sure what the target image should be. Override G_HW_ARCH with intended minimal CPU arch requirement and set + pass new G_RASPBIAN variable.
+ DietPi-PREP | raspi-copies-and-fills is only made for armhf hence skip it on 64-bit RPi images
+ DietPi-PREP | On RPi with Raspberry Pi OS (64-bit), armhf architecture is currently required to pull libraspberrypi packages and a few others. Keep armhf support until all expected packages have been ported to arm64.
+ DietPi-PREP | Prevent potential failure on u-boot install on C4 since it is pre-installed on the image but not yet present in the repo. The intention is to leave the package installed, if it was before, to allow automated upgrade to the actual C4 u-boot image once it is pushed to the repo.
+ DietPi-Config | Graphics driver: Add AMD graphics driver and firmware install to menu. ToDo: The Intel driver is recommended for pre~2007 chips only, this should be either made clear or the entry should be removed. Post~2007 Intel chips are better supported by native Xorg modesetting.
+ DietPi-Config | Display resolution: Add new RPi4 full KMS GL driver overlay. This has been added with Linux 5.4, which is at time of writing not yet released, but we'll add it now regardless to be prepared.
+ DietPi-Config | Audio options: Add new RPi sound card overlays
+ DietPi-Config | Minor coding and wording
+ DietPi-Globals | G_EXEC: Add info about Raspbian vs Debian RPi image to bug report printout
+ DietPi-Survey | Add new G_RASPBIAN to upload file on RPi, so we can watch the development of Raspberry Pi OS (64-bit) usage, or in theory other Debian images on RPi
+ DietPi-Survey_report | Add RPi Raspbian vs Debian stats
+ DietPi-Survey_report | Make table headers links to their ID for easier selection and references
+ DietPi-Config | Do not store none|N/A placeholder text in variables if it is used as inputbox default but not an actual valid value, instead start with an empty inputbox then.
+ DietPi-Config | Fix shown boot wait for network state
+ DietPi-Config | Align G_CHECK_URL timeout and attempts wording and value range with dietpi.txt info and what wget+ping commands allow.

MichaIng committed:
v6.31

+ CHANGELOG | Remove obsolete known issues

MichaIng committed:
v6.31

+ DietPi-PREP | Remove "pi" user autologin on Raspberry Pi OS (64-bit): https://github.com/MichaIng/DietPi/issues/3570#issuecomment-648988475

MichaIng committed:
v6.31

+ DietPi-Imager | Open interactive bash subshell for image content review/changes if user chooses to
+ DietPi-Imager | Wrap fsck into G_EXEC error handler to allow rerun in case of found errors, wrap more commands into G_EXEC to prevent unexpected states and increased verbosity

MichaIng committed:
v6.31

+ DietPi-Globals | Remove G_GET_NET* functions until concept has been finished, probably postpone to next release

MichaIng committed:
v6.31

+ DietPi-Set_software | NTP mirror: Fix gateway detection

MichaIng committed:
v6.31

+ DietPi-Boot | Tiny

MichaIng committed:
v6.31

+ DietPi-PREP | Ship full GPLv2 license text, being good practise

MichaIng committed:
v6.31

+ dietpi.txt | Allow to skip the interactive license dialog via new setting AUTO_SETUP_ACCEPT_LICENSE=1. This resolves an issue where firstrun setup is done non-interactively (AUTO_SETUP_AUTOMATED=1) and a non-input autostart options is chosen, e.g. Chromium kiosk mode with no keyboard attached: https://github.com/MichaIng/DietPi/issues/3326

MichaIng committed:
v6.31

+ DietPi-Login | Skip interactive license dialog if it has been accepted via new dietpi.txt setting: AUTO_SETUP_ACCEPT_LICENSE=1

MichaIng committed:
v6.31

+ dietpi.txt | Clarify AUTO_SETUP_HEADLESS setting

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-Login | A new setting has been added which allows to skip the interactive GPLv2 license agreement on first login. Add/set "AUTO_SETUP_ACCEPT_LICENSE=1" in (/boot/)dietpi.txt in which case we assume you read and agreed the license text at least once. Many thanks to @mrbluecoat for reporting the urgent use-case of Chromium kiosk mode autostart without keyboard attached: https://github.com/MichaIng/DietPi/issues/3326

MichaIng committed:
v6.31

+ DietPi-Software | TigerVNC: Workaround external bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=932499
+ DietPi-Software | TigerVNC: Missing netbase dependency has been resolved since Bullseye
+ DietPi-Software | VNC Server: Check for RealVNC/TigerVNC binaries explicitly, else exit with error
+ DietPi-Software | XRDP: Do not apply backports rules on Raspbian
+ DietPi-Software | QuiteRSS: Re-add install code, which was (accidentally?) removed/left commented with DietPi v127 commit: https://github.com/MichaIng/DietPi/commit/7cb41f3a72cdaa370b12c65f0d3e916ddacc91de

MichaIng committed:
v6.31

+ CHANGELOG | QuiteRSS: Re-added the years ago accidentally removed install code block. If you ever wondered why there was no QuiteRSS after selecting it for install, now there will be. If it is marked as installed already, run either "dietpi-software install 22" or simply "apt install quiterss" to (really) get it.
+ CHANGELOG | TigerVNC: Worked around an external bug where the VNC server crashes after client logout. Many thanks to @Joulinar for investigating this issue and providing the workaround: https://dietpi.com/phpbb/viewtopic.php?p=25285#p25285

MichaIng committed:
v6.31

+ DietPi-Patch | Avoid setcap error message by attempting capability removal only if any are present
+ DietPi-Patch | Reset QuiteRSS install state if the package has not actually been installed: https://github.com/MichaIng/DietPi/commit/49f960a7953f44e5571ff2dde68a93efca37ec03
+ DietPi-Patch | Fix obsolete FAHClient service removal

MichaIng committed:
v6.31

+ DietPi-Software | Tiny

MichaIng committed:
v6.31

+ DietPi-Patch | getcap always returns true, so we need to check for output instead

MichaIng committed:
v6.31 (#3620)

+ DietPi-Software | WireGuard: Enable support for all devices where the kernel module is shipped with the kernel package. This is e.g. true for all "current" Armbian images with 5.X kernel. WireGuard has been natively integrated into Linux since v5.6 hence this integration will be of raising importance and allows to skip DKMS and headers install.
+ DietPi-Software | WireGuard: Work around an issue with some Odroids where /lib/modules/<version>/build symlink is missing when headers are installed after the kernel package. If a matching kernel and headers dir pair is found but no symlink, it is created now manually: https://github.com/MichaIng/DietPi/issues/3577

MichaIng committed:
v6.31

+ DietPi-Software | TigerVNC: Fix LD_PRELOAD assignment

MichaIng committed:
v6.31

+ DietPi-Software | Minor wording/comments and shortened forum URLs

MichaIng committed:
v6.31

+ DietPi-Software | Desktops: Shared desktop entries/icons are hosted on GitHub for a while, hence remove the obsolete dietpi.com URL check.
+ DietPi-Software | MATE: Install the core package only, the extra package is a meta package that pulls (as dependency) only the base package which again pulls (as dependency) only the core package. Hence without explicitly pulling recommendations, we just install two meta packages + core and should instead install core only.

MichaIng committed:
v6.31

+ DietPi-Software | TigerVNC: Apply libunwind workaround via binary patch, which fixes calls from other origins like XRDP
+ DietPi-Software | RealVNC: Do not enable enterprise subscription service vncserver-virtuald. If users have enterprise subscription, they should disable our vncserver.service and enable vncserver-virtuald.service to switch from static VNC session on screen 1 to ondemand RealVNC session spawn.
+ DietPi-Software | RealVNC: Integrate vncserver-x11-serviced.service start into vncserver.service in shared mode. Having it running concurrently to a virtual VNC session makes it attaching to the virtual VNC session, which allows connecting to it on port 5900 (no screen :1 required) but the overhead overhead of ~20 MiB + possible confusion is not worth it.
+ DietPi-Software | VNC Servers: Apply vncpasswd only on TigerVNC since RealVNC by default uses UNIX user authentication and does not support the "-f" option at all.
+ DietPi-Software | VNC Servers: Minor uninstall enhancements

MichaIng committed:
v6.31

+ DietPi-Software | TigerVNC: Hide "dpkg-query -s" output
+ DietPi-Software | TigerVNC: Print "'Waiting for X11 to start" only one on shared mode startup, then wait silently. As of service type, subsequent prints are no caught anymore by journal and otherwise would spam it in case of disabled desktop autologin.

MichaIng committed:
v6.31

+ CHANGELOG | XRDP: The TigerVNC bug affected as well XRDP connections which have been resolved in the same turn: https://github.com/MichaIng/DietPi/issues/3615#issuecomment-650559035

MichaIng committed:
v6.31

+ Subversion up

MichaIng committed:
v6.31

+ DietPi-Globals | Default subversion up

MichaIng committed:
v6.31

+ DietPi-Patch | On Stretch, libcap2-bin ist no dependency of iproute2 and can hence be missing. Skip ping binary capability removal, if getcap/setcap is missing anyway. No need to force its install, security implication is minimal anyway.

MichaIng committed:
v6.31

+ DietPi-FirstBoot | Network setup: Although we do not ship our images with resolvconf anymore, be more compatible and apply static DNS nameservers correctly when it is present.

MichaIng committed:
v6.31

+ DietPi-Cleaner | Only mark packages for autoremoval if any were found
+ DietPi-Cleaner | Do not mark git as as not installed since we do not autoremove it anymore, since is required by some software titles for their internal updaters.

MichaIng committed:
v6.31

+ DietPi-Software | WiFi Hotspot: In case of RTL8188C* adapter on RPi, stay with default driver, since the issue seems to have been resolved differently and "rtl871xdrv" is not supported: https://github.com/MichaIng/DietPi/issues/2531#issuecomment-650810121

MichaIng committed:
Beta v6.31.0 (#3621)

Joulinar committed:
Update BRANCH_SYSTEM.md (#3629)

* Update BRANCH_SYSTEM.md

update /DietPi/dietpi.txt to /boot/dietpi.txt

* v6.31

+ BRANCH_SYSTEM.md | Typo

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng committed:
v6.31 (#3631)

+ Overclocking | Fix RPi3+ models detection

MichaIng committed:
v6.31

+ DietPi-Survey_report | Add Firefox Sync server and fix degree sign in html page

MichaIng committed:
v6.31

+ DietPi-FS_partition_resize | Use "DefaultDependencies=no" instead of "=false": https://github.com/systemd/systemd/commit/c13fb257343aaeaa5508152257b5c43cdf25cb49

MichaIng committed:
v6.31

+ DietPi-LED_control | Resolve an issue where LED triggers fail to become boot persistent if the related udev rules file does not yet exist

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-LED_control | Resolved an issue where LED triggers fail to become boot persistent. Many thanks to @FrostyMisa for reporting this issue: https://github.com/MichaIng/DietPi/issues/3636

MichaIng committed:
v6.31

+ DietPi-Config | Avoid exit code "1" when exiting dietpi-config the regular way without reboot

MichaIng committed:
v6.31

+ DietPi-Config | Fix current RPi memory split estimation for > 1024 MiB models: https://github.com/MichaIng/DietPi/issues/3636#issuecomment-652467249

MichaIng committed:
v6.31

+ CHANGELOG | DietPi-Config: Resolved an issue on RPi4 2/4/8 GiB RAM models where the current GPU memory split was not obtained correctly. Many thanks to @FrostyMisa for reporting this issue: https://github.com/MichaIng/DietPi/issues/3636#issuecomment-652467249

MichaIng committed:
v6.31

+ DietPi-FirstBoot | Update RPi overclocking value comments if NO manual changes have been done, instead of if such have been done

MichaIng committed:
v6.31

+ DietPi-WiFi-Monitor | Tiny

MichaIng committed:
v6.31 (#3635)

+ DietPi-Software | SqueezeLite: Install new self-compiled packages based upstream v1.9.7
+ DietPi-Software | SqueezeLite: Remove now obsolete systemd service hosted on GitHub. This will break pre-v6.31 installs once merged, however installing new software with outdated DietPi version is anyway not something we can reasonably support as installers require regular maintenance/update due to upstream changes anyway.
+ DietPi-Patch | Reinstall SqueezeLite to update to new v1.9.7
+ DietPi-Patch | If we already check /boot/dietpi/.installed for a certain install flag, add related software ID conditionally to reinstalls list to avoid unnecessary "XY not installed, nothing to do..." messages.
+ DietPi-Software | SqueezeLite: Skip download of test media since it is only a frontend for LMS which already pulls test media on install.
+ DietPi-Software | Logitech Media Server: If we already show previous names in software description, add the full list of names that were used for the SqueezeBox streaming server.
+ DietPi-Software | Logitech Media Server: Get latest 7.9 patch version (still 7.9.2 at time of writing)
+ DietPi-Software | Logitech Media Server: Do not add audio steaming server user to "render" group which is for DRM graphics device access only.

MichaIng committed:
v6.31

+ DietPi-Software | Logitech Media Server: Do not add user to "audio" group. Is does not access to audio devices directly but streams audio to SqueezeBox/SqueezeLite devices.

MichaIng committed:
v6.31

+ RC up

MichaIng committed:
v6.31

+ DietPi-Globals | Default RC up

MichaIng committed:
v6.31

+ DietPi-Patch | Do not remove /etc/systemd/system/squeezelite.service as it may contain custom user settings, instead just add/change running it as "squeezelite" user.

MichaIng committed:
v6.31

+ DietPi-Software | Change test media permissions to dietpi:dietpi 664 so all download/media software in "dietpi" group can (re)move it

MichaIng committed:
v6.31

+ CHANGELOG | Set release PR and date

MichaIng committed:
Beta v6.31.1 (#3639)

Joulinar commented on CHANGELOG.txt:
```diff
@@ -1,5 +1,5 @@
 v6.31
-(XX/06/20)
+(05/06/20)
```
I guess you would need to change date to 05.07.20

MichaIng replied:
Before I forget it: https://github.com/MichaIng/DietPi/commit/c7fd2a9b6904e2a531423311ad56ea1ddfb21034

MichaIng committed:
v6.31

+ CHANGELOG | Fix release date

MichaIng committed:
v6.31

+ DietPi-LED_control | Tiny consistency

MichaIng committed:
v6.31

+ DietPi-Launcher | Add DietPi-LED_control; Silence shellcheck about "read" without "-r" even that here we do not actually parse input

MichaIng committed:
v6.31 (#3644)

+ Coding | https://github.com/koalaman/shellcheck/wiki/SC2162: Since we do not scrape input but only hand to hold script execution via "-p", "-r" has no affect. But lets mute spellcheck.
+ Coding | https://github.com/koalaman/shellcheck/wiki/SC1087: Since we do not want to expand an array use square brackets for regex matching, curtly braces are not required. But lets mute spellcheck.

MichaIng committed:
v6.31

+ DietPi-Set_hardware | https://github.com/MichaIng/DietPi/pull/3644

MichaIng committed:
v6.31

+ DietPi-Patch | Copy correct license file: https://github.com/MichaIng/DietPi/issues/3628#issuecomment-653676563

MichaIng committed:
v6.31

+ DietPi-PREP | Remove all autologin configs for all TTYs: https://github.com/MichaIng/DietPi/issues/3628#issuecomment-653693758

MichaIng committed:
v6.31

+ DietPi-FirstBoot | Assure that root dir / permissions 755 root:root since this can be manipulated by flash method/program/user in rare cases: https://github.com/MichaIng/DietPi/issues/3646#issuecomment-653739919

MichaIng committed:
v6.31

+ RC up

MichaIng committed:
v6.31

+ DietPi-Globals | Default RC up

MichaIng committed:
v6.31

+ CHANGELOG | Minor spelling fixes

MichaIng committed:
v6.31

+ README.md | Remove /DietPi/ from files list, which is not used anymore

MichaIng committed:
v6.31

+ DietPi-Survey_report | Uploads untouched since 2020 have been removed to give a more current stat picture

MichaIng committed:
Beta v6.31.2 (#3648)
